plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -332,4 +345,50 @@ public static LocationProvider getLocationProvider(SchemaTableName schemaTableNa
         }
         return locationsFor(tableLocation, storageProperties);
     }
+
+    public static Schema toIcebergSchema(List<ColumnMetadata> columns)
+    {
+        List<Types.NestedField> icebergColumns = new ArrayList<>();
+        for (ColumnMetadata column : columns) {
+            if (!column.isHidden()) {
+                int index = icebergColumns.size();
+                org.apache.iceberg.types.Type type = toIcebergType(column.getType());
+                Types.NestedField field = column.isNullable()
+                        ? Types.NestedField.optional(index, column.getName(), type, column.getComment())
+                        : Types.NestedField.required(index, column.getName(), type, column.getComment());
+                icebergColumns.add(field);
+            }
+        }
+        org.apache.iceberg.types.Type icebergSchema = Types.StructType.of(icebergColumns);
+        AtomicInteger nextFieldId = new AtomicInteger(1);
+        icebergSchema = TypeUtil.assignFreshIds(icebergSchema, nextFieldId::getAndIncrement);
+        return new Schema(icebergSchema.asStructType().fields());
+    }
+
+    public static Transaction newCreateTableTransaction(TrinoCatalog catalog, ConnectorTableMetadata tableMetadata, ConnectorSession session)
+    {
+        SchemaTableName schemaTableName = tableMetadata.getTable();
+        Schema schema = toIcebergSchema(tableMetadata.getColumns());
+        PartitionSpec partitionSpec = parsePartitionFields(schema, getPartitioning(tableMetadata.getProperties()));
+        String targetPath = getTableLocation(tableMetadata.getProperties()).orElse(catalog.defaultTableLocation(session, schemaTableName));
+        ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builderWithExpectedSize(2);
+        FileFormat fileFormat = IcebergTableProperties.getFileFormat(tableMetadata.getProperties());
+        propertiesBuilder.put(DEFAULT_FILE_FORMAT, fileFormat.toString());
+        if (tableMetadata.getComment().isPresent()) {
+            propertiesBuilder.put(TABLE_COMMENT, tableMetadata.getComment().get());
+        }
+
+        return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, targetPath, propertiesBuilder.build());
+    }
+
+    public static boolean isHiveMaterializedView(io.trino.plugin.hive.metastore.Table table)
+    {
+        return isHiveView(table) && table.getParameters().containsKey(STORAGE_TABLE);
+    }
+
+    public static boolean isHiveView(io.trino.plugin.hive.metastore.Table table)
```

Comment:
I suggest moving this method to `HiveTableOperations` for now, since it's only used there. Similarly, move `isTrinoMaterializedView` to `TrinoHiveCatalog`. Probably inline them since they're only used in one place, and use the static methods from `ViewReaderUtil` if possible.

Reply:
sure, I will change the name and call the `ViewReaderUtil`